app/public/js/common/configLocation.js

```diff
@@ -1,36 +1,61 @@
 const fs = require('fs-extra');
 const userHome = require('user-home');
+const mkdirp = require('mkdirp');
+
+/**
+ * Check if a given folder exists before trying to access any of it's children
+ * 
+ * @param {String} folder The folder we're checking
+ */
+const folderExists = folder => {
+    let exists = false;
+    fs.stat(folder, (err, stats) => { if (stats.isDirectory()) exists = true; });
```

Comment:
Use `fs.statSync` to avoid race conditions since Node is async and we are returning `exists` before this is done.